File: web/backend/schemas/node.py
"""
节点相关数据模型
"""
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, Field


class NodeCreateRequest(BaseModel):
    """创建节点请求"""
    node_name: str = Field(..., min_length=1, max_length=100, description="节点名称")
    platform: str = Field(..., pattern="^(linux|windows)$", description="平台类型")
    description: Optional[str] = Field(None, max_length=500, description="节点描述")


class NodeResponse(BaseModel):
    """节点响应"""
    id: int
    node_name: str
    virtual_ip: str
    public_key: str
    platform: str
    description: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class NodeDetailResponse(NodeResponse):
    """节点详情响应（包含私钥）"""
    private_key: Optional[str] = None

File: web/frontend/src/types/node.ts
/**
 * 节点相关类型定义
 */

// 节点平台类型
export type NodePlatform = 'linux' | 'windows'

// 创建节点请求参数
export interface NodeCreateRequest {
  node_name: string
  platform: NodePlatform
  description?: string
}

// 节点响应（列表）
export interface Node {
  id: number
  node_name: string
  virtual_ip: string
  public_key: string
  platform: NodePlatform
  description: string | null
  created_at: string | null
  updated_at: string | null
}

// 节点详情响应（包含私钥）
export interface NodeDetail extends Node {
  private_key: string | null
}
